tox.ini

```diff
@@ -0,0 +1,81 @@
+[tox]
+envlist =
+    py{36,37,38}-test{,-alldeps,-devdeps}
+    build_docs
+    linkcheck
+    style
+requires =
+    setuptools >= 30.3.0
+    pip >= 19.3.1
+isolated_build = true
+
+[testenv]
+passenv =
+    HOME
+    WINDIR
+    LC_ALL
+    LC_CTYPE
+    CC
+    CFLAGS
+    CI
+    TRAVIS
+    TEST_READ_HUGE_FILE
+changedir =
+    test: .tmp/{envname}
+    build_docs,linkcheck: docs
+description =
+    test: run tests with pytest
+    build_docs: invoke sphinx-build to build the HTML docs
+    linkcheck: check the links in the HTML docs
+    alldeps: run tests with all dependencies
+    devdeps: run tests with numpy and matplotlib dev versions
+deps =
+    # The oldestdeps tag is intended to be used to install the oldest versions of all
+    # dependencies that have a minimum version
+    oldestdeps: numpy==1.16.*
+    oldestdeps: matplotlib==2.1.*
+    oldestdeps: asdf==2.4.*
+    numpy116: numpy==1.16.*
+    numpy117: numpy==1.17.*
+    numpy118: numpy==1.18.*
+    devdeps: git+https://github.com/numpy/numpy.git#egg=numpy
+    devdeps: git+https://github.com/matplotlib/matplotlib.git#egg=matplotlib
+extras =
+    test: test,cov
+    build_docs,linkcheck: docs
+    alldeps: all
+commands =
+    pip freeze
+    test: pytest --pyargs astropy {toxinidir}/docs --cov astropy --cov-config={toxinidir}/astropy/tests/coveragerc {posargs}
+    build_docs: sphinx-build -W -b html . _build/html {posargs}
+    linkcheck: sphinx-build -W -b linkcheck . _build/html {posargs}
```

Comment:
For readability I would put `build_docs` and `linkcheck` in their own section, otherwise it's a bit difficult to find all the pieces. Something like
```
 [testenv:build_docs]
changedir = docs
description = invoke sphinx-build to build the HTML docs
extras = docs
commands = sphinx-build -W -b html . _build/html {posargs}
```
is much clearer imo, and easier to copy/reuse/modify.

Reply:
Done